src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/documentation/LSPTextHoverForFile.java

```diff
@@ -52,15 +54,16 @@ public List<MarkupContent> getHoverContent(PsiElement element, int targetOffset,
             // The LSP hover request are finished, don't need to cancel the previous LSP requests.
             previousCancellationSupport = null;
             return result;
-        } catch (ExecutionException e) {
-            if (!(e.getCause() instanceof CancellationException)) {
+        } catch (ResponseErrorException | ExecutionException | CancellationException e) {

```

Comment:
shouldn't CancellationExceptions be rethrown?

Reply:
I don't think so, because I think IJ will log this error.

Reply:
so throw it as ProcessCanceledException?

Reply:
To be honnest with you, I don't know how IJ documentation is working and I don't know if ProcessCanceledException will be logged as an error or used for a particular behavior (ex: retry the hover).

My PR should keep the current behavior. We could investigate more in a separate PR i fthrowing ProcessCanceledException could be interesting.